src/Services/PaymentCreation.php

```diff
@@ -133,6 +134,26 @@ public function createPayment($mopId, ResponseAbstract $response, Basket $basket
             $payment->receivedAt = date('Y-m-d H:i:s');
         }
 
+        try {
+            /** @var CurrencyExchangeRepositoryContract $currencyService */
+            $currencyService = pluginApp(CurrencyExchangeRepositoryContract::class);
+
+            $defaultCurrency = $currencyService->getDefaultCurrency();
+
+            if ($payment->currency != $defaultCurrency) {
+                $payment->exchangeRatio = $currencyService->getExchangeRatioByCurrency($payment->currency);
+                $payment->amount = round(
+                    $currencyService->convertToDefaultCurrency(
+                        $payment->currency,
+                        $payment->amount,
+                        $payment->exchangeRatio
+                    ),
+                    2);
+            }
+        } catch (\Exception $cEx) {
```

Comment:
Why is this necessary? These empty catch expressions are always a nightmare for debugging.

Reply:
solved